raiden/accounts.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+import os
+import sys
+import json
+import getpass
+
+from pyethapp.accounts import Account
+
+
+def find_datadir():
+    home = os.path.expanduser("~")
+    datadir = None
+    if sys.platform.startswith('linux'):
+        datadir = os.path.join(home, '.ethereum')
+    elif sys.platform == 'darwin':
+        datadir = os.path.join(home, 'Library', 'Ethereum')
+    else:
+        raise RuntimeError('Unsupported Operating System')
+
+    if os.path.isdir(datadir) is False:
+        return None
+    return datadir
+
+
+def find_keystoredir():
+    datadir = find_datadir()
+    if datadir is None:
+        # can't find a data directory in the system
+        return None
+    keystore_path = os.path.join(datadir, 'keystore')
+    if not os.path.exists(keystore_path):
+        # can't find a keystore under the found data directory
+        return None
+    return keystore_path
+
+
+class AccountManager:
+
+    def __init__(self, keystore_path=None):
+        self.accounts = {}
+        self.keystore_path = keystore_path
+
+    def get_accounts(self):
+        """Find all the accounts the user has locally and return their addresses
+
+        :return dict: A dictionary with the addresses of the accounts of the user
+                      as the key and the filenames as the value
+        """
+        if self.accounts:
```

Comment:
Desaster is a bit harsh. But it using account_manager.accounts may give you the wrong result if it hasn't been initialized. I know, in our case it was initialized, but there's really no need to give the developer the chance to use the class in a wrong way.

Anyway, it's fixed with the initialization being moved to __init__